cmd/group-auto-updater/main_test.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	coretesting "k8s.io/client-go/testing"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/diff"
+
+	v1 "github.com/openshift/api/user/v1"
+	"github.com/openshift/client-go/user/clientset/versioned/fake"
+)
+
+func TestUpdateGroup(t *testing.T) {
```

Comment:
I think this test is still trying to accomplish too much - it's a hybrid between a unit test and an integration test, and the result is that it tests multiple things and is unnecessarily complicated. Any time when a **unit** test needs to hit a filesystem, it is usually a very good indicator that some refactoring is in order.

The tool essentially does three things: 

1. Reads Peribolos config: we do not need to unit-test this (we only use other packages for this)
2. From Peribolos config read a list of logins to put into the group. This is very simple functionality and we probably would do without unit testing that (it is just reading two fields from the peribolos config), although it's easy and would not hurt.
3. Take a list of logins, a group and a client, and update the group accordingly. This is the `updateGroup` method and it should definitely be unit tested. For unit testing it, you do not even need Peribolos config: the method's input is just a list of logins. Thorough unit tests would also test error scenarios like that the method can handle error conditions gracefully.

In addition to unit tests, I think at least one integration test is always good - we have these in bash and over real(ish) input - like @stevekuznetsov [suggested](https://github.com/openshift/ci-tools/pull/300#pullrequestreview-323962872), we should use the dry-logger here to check that the tools issues expected API calls.

Reply:
Yea integration tests using the drylogger would be nice. I will add them in this PR, and I will make the unit tests simplier